https://github.com/RadeonOpenCompute/ROCm-CompilerSupport/pull/25

From c65cba2e73f9118e128b9ab7e655ee0f8a7798e7 Mon Sep 17 00:00:00 2001
From: Craig Andrews <candrews@integralblue.com>
Date: Sun, 1 Mar 2020 19:24:22 -0500
Subject: [PATCH] Link additional required LLVM libraries

Without these additional required dependencies, linking fails with errors such as:
`undefined reference to llvm::errs()'`
---
 CMakeLists.txt | 20 ++++++++++++++++++--
 1 file changed, 18 insertions(+), 2 deletions(-)

diff --git a/lib/comgr/CMakeLists.txt b/lib/comgr/CMakeLists.txt
index 8b5ca2f..a7d226f 100644
--- a/lib/comgr/CMakeLists.txt
+++ b/lib/comgr/CMakeLists.txt
@@ -294,7 +294,11 @@ install(FILES
   DESTINATION "${AMD_COMGR_PACKAGE_PREFIX}")
 
 set(CLANG_LIBS
-  clangFrontendTool)
+  clangFrontendTool
+  clangFrontend
+  clangBasic
+  clangDriver
+  clangSerialization)
 
 set(LLD_LIBS
   lldELF
@@ -305,8 +309,21 @@ if (LLVM_LINK_LLVM_DYLIB)
 else()
   llvm_map_components_to_libnames(LLVM_LIBS
     ${LLVM_TARGETS_TO_BUILD}
+    Option
     DebugInfoDWARF
-    Symbolize)
+    Symbolize
+    Support
+    Object
+    BitWriter
+    MC
+    MCParser
+    MCDisassembler
+    Core
+    IRReader
+    CodeGen
+    Linker
+    Demangle
+    BinaryFormat)
 endif()
 
 target_link_options(amd_comgr
